packages/aragon-cli/src/commands/apm_cmds/publish.js

```diff
@@ -107,8 +107,50 @@ exports.builder = function(yargs) {
     })
 }
 
+const getMajor = version => version.split('.')[0]
+
+function getFunctionsAbi(functions, abi) {
+  functions.forEach(f => {
+    f.abi = abi.find(method => method.name === f.sig.match(/(.*)\(/)[1])
+  })
+}
+
+async function getDeprecatedFunctions(apm, artifact) {
+  let deprecated = {}
+  try {
+    let deprecatedOnVersion = []
+    const deprecatedFunctionsSig = []
+    const versions = await apm.getAllVersions(artifact.appName)
+    let lastMajor = -1
+    versions.reverse().forEach(async repo => {
+      // iterate on major versions
+      if (getMajor(repo.version) !== lastMajor) {
+        lastMajor = getMajor(repo.version)
+        repo.functions.forEach(f => {
+          if (
+            !artifact.functions.some(obj => obj.sig === f.sig) &&
+            !deprecatedFunctionsSig.includes(f.sig)
+          ) {
+            deprecatedOnVersion.push(f)
+            deprecatedFunctionsSig.push(f.sig)
+          }
+        })
+        if (deprecatedOnVersion.length) {
+          deprecated[`${lastMajor}.0.0`] = deprecatedOnVersion
+          getFunctionsAbi(deprecatedOnVersion, repo.abi)
+          deprecatedOnVersion = []
+        }
+      }
+    })
+    return deprecated
```

Comment:
I think it'd be good to move this return to the end of the function, so this function always returns an object.